model/map/src/main/java/org/keycloak/models/map/clientscope/MapClientScopeAdapter.java

```diff
@@ -16,25 +16,32 @@
  */
 package org.keycloak.models.map.clientscope;
 
+import java.util.Collection;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Objects;
+import java.util.Set;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import org.keycloak.models.ClientScopeModel;
 import org.keycloak.models.KeycloakSession;
 import org.keycloak.models.ProtocolMapperModel;
 import org.keycloak.models.RealmModel;
 import org.keycloak.models.RoleModel;
+import org.keycloak.models.map.client.MapProtocolMapperEntity;
+import org.keycloak.models.map.client.MapProtocolMapperUtils;
 import org.keycloak.models.utils.KeycloakModelUtils;
 import org.keycloak.models.utils.RoleUtils;
 
 public class MapClientScopeAdapter extends AbstractClientScopeModel<MapClientScopeEntity> implements ClientScopeModel {
 
+    private final MapProtocolMapperUtils pmUtils;
+
     public MapClientScopeAdapter(KeycloakSession session, RealmModel realm, MapClientScopeEntity entity) {
         super(session, realm, entity);
+        pmUtils = new MapProtocolMapperUtils(safeGetProtocol());
```

Comment:
Wondering - could this be 
```suggestion
        pmUtils = MapProtocolMapperUtils.instanceFor(safeGetProtocol());
```
where `MapProtocolMapperUtils.instanceFor` would leverage a static `Map<String, MapProtocolMapperUtils>` that would be cache previously created instances for a particular protocol?

Reply:
Thank you, that way there would be only two instances created.